src/gt4py/next/iterator/transforms/constant_folding.py

```diff
@@ -20,38 +20,129 @@ def visit_FunCall(self, node: ir.FunCall):
+                        if fun_call.fun.id == "plus":
+                            new_node = sym_ref if int(fun_call.args[1].value) >= 0 else fun_call
+                        elif fun_call.fun.id == "minus":
+                            new_node = fun_call if int(fun_call.args[1].value) > 0 else sym_ref
+                    elif new_node.fun.id == "maximum":
+                        if fun_call.fun.id == "plus":
+                            new_node = fun_call if int(fun_call.args[1].value) > 0 else sym_ref
+                        elif fun_call.fun.id == "minus":
+                            new_node = sym_ref if int(fun_call.args[1].value) >= 0 else fun_call
+        # `if_(True, true_branch, false_branch)` -> `true_branch`
+        if cpm.is_call_to(new_node, "if_") and isinstance(
+            new_node.args[0], ir.Literal
+        ):
             if new_node.args[0].value == "True":
                 new_node = new_node.args[1]
             else:
                 new_node = new_node.args[2]
-
+        # `1 + 1` -> `2`
         if (
             isinstance(new_node, ir.FunCall)
             and isinstance(new_node.fun, ir.SymRef)
             and len(new_node.args) > 0
             and all(isinstance(arg, ir.Literal) for arg in new_node.args)
-        ):  # `1 + 1` -> `2`
+        ):
             try:
                 if new_node.fun.id in ir.ARITHMETIC_BUILTINS:
                     fun = getattr(embedded, str(new_node.fun.id))
                     arg_values = [
-                        getattr(embedded, str(arg.type))(arg.value)  # type: ignore[attr-defined] # arg type already established in if condition
+                        getattr(embedded, str(arg.type))(arg.value)
+                        # type: ignore[attr-defined] # arg type already established in if condition
                         for arg in new_node.args
                     ]
                     new_node = im.literal_from_value(fun(*arg_values))
             except ValueError:
                 pass  # happens for inf and neginf
-
+        # `__out_size_1 + 1 + 1` -> `__out_size_1 + 2`
+        if (
+            isinstance(new_node, ir.FunCall)
+            and isinstance(new_node.fun, ir.SymRef)
+            and len(new_node.args) == 2
+        ):
+            match = False
+            if isinstance(new_node.args[0], ir.FunCall) and isinstance(
+                new_node.args[1], ir.Literal
+            ):
+                fun_call, literal = new_node.args
+                match = True
+            elif isinstance(new_node.args[0], ir.Literal) and isinstance(
+                new_node.args[1], ir.FunCall
+            ):
+                match = True
+                literal, fun_call = new_node.args
+            if match and cpm.is_call_to(fun_call, ("plus", "minus")):
+                new_node = im.plus(
+                    fun_call.args[0],
+                    ConstantFolding.apply(im.call(fun_call.fun.id)(fun_call.args[1], literal)),
```

Comment:
A cleaner solution is to call `self.visit` again instead of initializing the pass again. The fixed point transformation is even better suited here.